### ISSUE TITLE
Presentation by Maxime Paillassa 

### ISSUE THREAD

**Issue by sfarrens** [CosmoStat/Shear-and-PSF-Reading-Group#49]
# Robust detection of astronomical sources using convolutional neural networks

> Speaker: Maxime Paillassa 
> Date: 29/04/2021
> Slides: [PDF](https://drive.google.com/file/d/1h5q7-PYBVrRK-bIk73to9wfyLJIK7-8S/view?usp=sharing)

**Comment by andrevitorelli** [comment]
Quick notes: 

cnn for robust detection

Maxime Paillassa
- astronomers usually work on catalogues
- problems: deblending, nearby stars
- robust: identify contaminants (image defects)
- Context: Cosmic-DANCe (IMF of stars)
- Heterogenous data, with proper motions...
- Context2: Euclid
- Huge data volume space/ground images (matching bc euclid photometry not enough for z-ph)
- Sextractor++
- current: several steps: bkg subtraction, matched filter, peaksearch/thresh/deblending
- matched: correlation signal/pattern
- optimal linear filter -> mas SNR 
- problems: sources have various scales, can overlap
- matched filter detects one pattern at a time -> problems
- use NN
- huge amounts of data available
- image -> semantic segmentation -> instance(-aware semantic) segmentation
- Mask R-CNN: boxes, rescaling implies interpolations, no deblending, needs clear boundaries
- multiscale footprint segmentation
- brightest pixels, pixels above a fraction, compromise, configurable

Alexandre:
- def is relative to max intensity, no relation to the noise?

Maxime:
- yes, on noise-free, isolated sources (training) -> SkyMaker (Bertin 2006)

- assign a scale to sources
- galaxies (training): cosmo sims, stars SkyMaker
- train images are contaminated to train robustness

Sam:
- statistics over SX?
Maxime:
- Over test sets.

Bastien:
- plan to compare ground/space telescope?
Maxime:
- Good point HST images as ground truth
Bastien:
- multiband detection? how to manage footprint different detection
Alexandre:
- as per definition, I think there wouldn't be much of a difference between filters

Axel:
- what is the actual output: pos of peak? more?
Maxime:
- footprint map probability of pixel being part of detection
Axel:
- how to combine scales together?
Maxime:
- for now not much confusion. postprocessing
Axel:
- here, for example (slide 19): how do you plot the objects?
Maxime:
- combination of all scales
Axel:
- would we then not need masking?
Maxime:
- that's one of point of this, yes
Sam:
- how do you trust when it's sensitive to the detection
Maxime:
- this is tricky, more simulations to quantify
Axel:
- q related: more sources CNN, properties of that sources: are they low SNR? gain from deblending?
Maxime:
- cnn can really find fainter sources, apparently
Alexandre:
- what is sextractor here, because you can tune it in many ways
Maxime:
- this example, gaussian kernel (5x5), 1.5 x sigma, no hot/cold approach
Alexandre/Maxime:
- people will say that they can run sextractor better, the NN is not tunned.

Alexandre:
- speed?
Maxime:
- >1mpixel/sec + preprocessing
Sam:
- important is if you can process your large area

Alexandre/Maxime:
- train on noiseless, and then in noisy? Noiseless footprint, but training on noisy

Alexandre:
- NN has to be trained on different noises?
Maxime:
- now maybe the range is not enough, but we could expand

Alexandre:
- could compute completeness vs snr
- it seems that footprints are smaller
- Deblending? multiple scale approach is good, but also obj can have star forming regions... 

Axel:
- bkg: how sensitive if bkg is not well subtracted? can the network avoid bkg subtraction?
Maxime:
- i think it's necessary. i didn't include very bad bkg, but yes variable... it should be able to detect in this conditions. 
- the one that I use is same as sx.
Axel:
- last q: varying psf not smoothly, is that a problem, like, in coadds -> variation of the size of stars. would that be a problem to the CNN
Maxime:
- not tested, future simluation.
